locator/tablets.cc

```diff
@@ -430,6 +430,36 @@ class tablet_effective_replication_map : public effective_replication_map {
         return result;
     }
 
+    std::optional<tablet_routing_info> check_locality(const token& search_token) const override {
+        auto&& tablets = get_tablet_map();
+        auto tid = tablets.get_tablet_id(search_token);
+        auto&& info = tablets.get_tablet_info(tid);
+        auto host = get_token_metadata().get_my_id();
+        auto shard = this_shard_id();
+
+        auto make_tablet_routing_info = [&] {
+            auto token_range = std::make_pair(tablets.get_first_token(tid), tablets.get_last_token(tid));
```

Comment:
I wonder if it wouldn't be easier for the driver if the range was [x, y), with last token excluded, so that it's equal to the first token of the next range. This way the adjacency check on the driver doesn't require manipulating with the token value.

Reply:
That way here I would have to do manipulation? I can change that, but I am not sure if it really makes a difference for the driver

Reply:
Agree semi-open ranges are better. It means that if you lay out all the tablets, the right edge of one is equal to the left edge of the next.

Reply:
Okay, I changed that